javascript/packages/linter/src/rules/html-no-empty-attributes.ts

```diff
@@ -0,0 +1,78 @@
+import { ParserRule } from "../types.js"
+import { AttributeVisitorMixin, StaticAttributeStaticValueParams, DynamicAttributeStaticValueParams } from "./rule-utils.js"
+
+import type { LintOffense, LintContext } from "../types.js"
+import type { ParseResult } from "@herb-tools/core"
+
+// Attributes that must not have empty values
+const RESTRICTED_ATTRIBUTES = new Set([
+  'id',
+  'class',
+  'name',
+  'for',
+  'src',
+  'href',
+  'alt',
+  'title',
+  'data',
+  'role'
+])
+
+// Check if attribute name matches any restricted patterns
+function isRestrictedAttribute(attributeName: string): boolean {
+  const lowerName = attributeName.toLowerCase()
```

Comment:
I think the `attributeName`  that we pass from the `AttributeVisitorMixin` is already lowercased:

https://github.com/marcoroth/herb/blob/b2029f2c9134f3b025639f1412f358c0fbedcd26/javascript/packages/linter/src/rules/rule-utils.ts#L188